hack/check-conformance-test-requirements.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This tool is for checking conformance e2e tests follow the requirements
+// which is https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md#conformance-test-requirements
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"regexp"
+)
+
+var exitCode = 0
+
+// This function checks the requirement: it works for all providers (e.g., no SkipIfProviderIs/SkipUnlessProviderIs calls)
+func checkAllProviders(e2eFile string) {
+	inConformanceCode := false
+
+	//e.g. framework.ConformanceIt("should provide secure master service ", func() {
+	regStartConformance := regexp.MustCompile("framework.ConformanceIt\\(.*, func\\(\\) {$")
+	regEndConformance := regexp.MustCompile("}\\)$")
+	regSkipProviderIs := regexp.MustCompile("Skip.*ProviderIs\\(")
+
+	fileInput, err := ioutil.ReadFile(e2eFile)
+	if err != nil {
+		fmt.Println("Failed to read file %s: %v", e2eFile, err)
+		os.Exit(1)
+	}
+	scanner := bufio.NewScanner(bytes.NewReader(fileInput))
+	scanner.Split(bufio.ScanLines)
+
+	for scanner.Scan() {
+		line := scanner.Text()
+		if regStartConformance.MatchString(line) {
+			if inConformanceCode {
+				panic("Missed the end of previous conformance test. There might be a bug in this script.")
+			}
+			inConformanceCode = true
+		}
+		if inConformanceCode {
+			if regSkipProviderIs.MatchString(line) {
+				fmt.Printf("%v: Conformance test should not call SkipIfProviderIs()/SkipUnlessProviderIs()\n", e2eFile)
+				exitCode = 1
```

Comment:
You could just return error from this function and any error = fail.

Reply:
I was thinking it would be nice to list all invalid places on one operation.